migrator/migrations/m_109_to_m_110_replace_deprecated_resources/migration.go

```diff
@@ -0,0 +1,134 @@
+	ImageComponent       = "ImageComponent"
+	ImageIntegration     = "ImageIntegration"
+	Indicator            = "Indicator"
+	Licenses             = "Licenses"
+	NetworkBaseline      = "NetworkBaseline"
+	Notifier             = "Notifier"
+	ProcessWhitelist     = "ProcessWhitelist"
+	Risk                 = "Risk"
+	SignatureIntegration = "SignatureIntegration"
+	User                 = "User"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 109,
+		VersionAfter:   storage.Version{SeqNum: 110},
+		Run: func(databases *types.Databases) error {
+			return migatePermissionSets(databases.RocksDB)
+		},
+	}
+
+	prefix = []byte("permission_sets")
+
+	replacements = map[string]string{
+		APIToken:             Integration,
+		AuthProvider:         Access,
+		BackupPlugins:        Integration,
+		Group:                Access,
+		ImageComponent:       Image,
+		ImageIntegration:     Integration,
+		Indicator:            DeploymentExtension,
+		Licenses:             Access,
+		NetworkBaseline:      DeploymentExtension,
+		Notifier:             Integration,
+		ProcessWhitelist:     DeploymentExtension,
+		Risk:                 DeploymentExtension,
+		SignatureIntegration: Integration,
+		User:                 Access,
+	}
+
+	readOpts  = gorocksdb.NewDefaultReadOptions()
+	writeOpts = gorocksdb.NewDefaultWriteOptions()
+)
+
+func init() {
+	migrations.MustRegisterMigration(migration)
+}
+
+func propagatePermission(resource string, accessLevel storage.Access, permissions map[string]storage.Access) storage.Access {
+	if _, found := permissions[resource]; !found {
+		return accessLevel
+	}
+	oldLevel := permissions[resource]
+	if accessLevel > oldLevel {
+		return oldLevel
+	}
+	return accessLevel
+}
+
+func migatePermissionSets(db *gorocksdb.DB) error {
+	it := db.NewIterator(readOpts)
+	defer it.Close()
+	wb := gorocksdb.NewWriteBatch()
+	defer wb.Destroy()
+	for it.Seek(prefix); it.ValidForPrefix(prefix); it.Next() {
+		permissions := &storage.PermissionSet{}
+		if err := proto.Unmarshal(it.Value().Data(), permissions); err != nil {
+			return errors.Wrap(err, "unable to unmarshal permission set")
+		}
+		// Copy the permission set, removing the deprecated resource permissions, and keeping the
+		// lowest access level between that of deprecated resource and their replacement
+		// for the replacement resource.
+		newPermissionSet := &storage.PermissionSet{
+			Id:               permissions.GetId(),
+			Name:             permissions.GetName(),
+			Description:      permissions.GetDescription(),
+			ResourceToAccess: make(map[string]storage.Access, len(permissions.GetResourceToAccess())),
+		}
+		for resource, accessLevel := range permissions.GetResourceToAccess() {
+			newResource := resource
+			if _, found := replacements[resource]; found {
+				newResource = replacements[resource]
+			}
+			newPermissionSet.ResourceToAccess[newResource] =
+				propagatePermission(newResource, accessLevel, newPermissionSet.ResourceToAccess)
```

Comment:
nit: I find the usage of `newResource` and `resource` a bit confusing when reading through the function, so maybe reduce the cognitive overhead and just keep `resource`.
```suggestion
			if _, found := replacements[resource]; found {
				resource = replacements[resource]
			}
			newPermissionSet.ResourceToAccess[resource] =
				propagatePermission(resource, accessLevel, newPermissionSet.ResourceToAccess)
```